strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryStreamSupport.java

```diff
@@ -125,6 +156,14 @@ protected void beforeWrite(int n)
             super.beforeWrite(n);
         }
 
+        @Override
+        public void flush()
+            throws IOException
+        {
+            RepositoryStreamSupport.this.commit();
```

Comment:
Are you sure `commit` should be called in `flush` rather than `close` ? `Flush` indicates we want to clear some buffer and it forces us to call it explicitly. We should benefit from `The try-with-resources Statement`

Reply:
yes, I am sure :)
it's good that you ask it because the thing is very interesting
using `try-with-resources` you can't be sure that there were no exceptions when `close()` called, so we need to expectedly call  `flush()` at the end of `try` statement.